app.rb

```diff
@@ -43,6 +43,19 @@
 end
 
 #
+#  Fetches a random definition by following the /random.php redirect
+#  and taking out the term with a regexp
+#
+get '/random' do
+  rdef = ud.get_random_definition()
+    unless valid_response? definition
+    raise NoDefinition, params[:term]
+  end
+
+  jsonp definition
```

Comment:
I'll be waiting, about the tests, I'll be glad to help as long as I can learn how to write them!